### DISCUSSION TITLE
Why does the field in Compound Unique Input become `Date | string` rather than `Date | null` when I add a Datetime? type field to the unique constraint?

### DISCUSSION THREAD

**Discussion by mmmmicha** [prisma/prisma#24825] [Q&A] [answered]
### Question

Hi, there?

I had to bind three columns with a unique constraint. At this time, I included a field called ‘deletedAt(type: `Datetime?`)’ in the compound input (to distinguish between null and non-null values). But something unexpected happened. Compound unique input was created as shown in the following picture, but deletedAt is `Date | string`, not `Date | null`. Hmm... I'm a little confused, but I'd like you to answer two questions.

    1. For what purpose did you type string instead of null?
    2. ‘What’ string should I put in the compound unique input created like this to search for a null value?

I would like to ask for a kind response!

Thank you.

### How to reproduce (optional)

```ts
// schema.prisma
model Test {
  /* some fileds */
  name String
  email String
  deletedAt Datetime? @db.Timestamptz(3)

  @@unique([name, email, deletedAt])
}

...
// index.d.ts
export type TestNameEmailDeletedAtCompoundUniqueInput = {
  name: string
  email: string
  deletedAt: Date | string // not Date | null
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @mmmmicha 👋 

Thank you for raising this question.

> 1. For what purpose did you type string instead of null?

This is done because NULL in SQL is not unique. This was discussed in this [issue](https://github.com/prisma/prisma/issues/22016). We also have an open [feature request](https://github.com/prisma/prisma/issues/3197) to support this behaviour. Do take a look at both issues.

> 2. What string should I put in the compound unique input created like this to search for a null value?

In the compound unique input, you should use `null` directly instead of a string when you want to search for a `null` value.
